packages/flutter_version_resolver/test/flutter_version_resolver_test.dart

```diff
@@ -0,0 +1,130 @@
+import 'dart:io';
+
+import 'package:flutter_version_resolver/flutter_version_resolver.dart';
+import 'package:flutter_version_resolver/src/logger.dart';
+import 'package:mason_logger/mason_logger.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:path/path.dart' as p;
+import 'package:pub_semver/pub_semver.dart';
+import 'package:scoped_deps/scoped_deps.dart';
+import 'package:test/test.dart';
+
+class _MockLogger extends Mock implements Logger {}
+
+void main() {
+  late Logger logger;
+  late Directory packageDirectory;
+  late File pubspecFile;
+
+  R runWithOverrides<R>(R Function() body) {
+    return runScoped(
+      body,
+      values: {loggerRef.overrideWith(() => logger)},
+    );
+  }
+
+  setUp(() {
+    logger = _MockLogger();
+
+    packageDirectory = Directory.systemTemp.createTempSync(
+      'flutter_version_resolver_test',
+    );
+    pubspecFile = File(
+      p.join(packageDirectory.path, 'pubspec.yaml'),
+    )..writeAsStringSync('name: flutter_version_resolver_test');
+  });
+
+  group('resolveFlutterVersion', () {
+    group('when no flutter version is specified in the pubspec.yaml file', () {
+      test('returns the stable version', () {
+        runWithOverrides(() {
+          expect(
+            resolveFlutterVersion(packagePath: packageDirectory.path),
+            equals('stable'),
+          );
+        });
+      });
+    });
+
+    group('when a flutter version is specified in the pubspec.yaml file', () {
```

Comment:
should we add a case for `when a flutter version range is specified`

Reply:
You mean like on line 92? :P

Reply:
yeah but we never explicitly cover `resolveFlutterVersion` with a version range as far as I can tell.